test/test_base_metric.py

```diff
@@ -7,84 +8,103 @@
 from test.test_utils import ids_metric_learners, metric_learners
 
 
+def remove_spaces(s):
+  return re.sub('\s+', '', s)
+
+
 class TestStringRepr(unittest.TestCase):
 
   def test_covariance(self):
-    self.assertEqual(str(metric_learn.Covariance()),
-                     "Covariance(preprocessor=None)")
+    self.assertEqual(remove_spaces(str(metric_learn.Covariance())),
+                     remove_spaces("Covariance(preprocessor=None)"))
 
   def test_lmnn(self):
     self.assertRegexpMatches(
-        str(metric_learn.LMNN()),
-        r"(python_)?LMNN\(convergence_tol=0.001, k=3, learn_rate=1e-07, "
-        r"max_iter=1000,\n      min_iter=50, preprocessor=None, "
-        r"regularization=0.5, use_pca=True,\n      verbose=False\)")
+      str(metric_learn.LMNN()),
+      r"(python_)?LMNN\(convergence_tol=0.001, k=3, learn_rate=1e-07, "
+      r"max_iter=1000,(\s+)min_iter=50, preprocessor=None, "
+      r"regularization=0.5, use_pca=True,(\s+)verbose=False\)")
```

Comment:
Don't need to wrap the whitespace matchers in parens.

Reply:
That's right, done